packages/event-emitter/src/index.ts

```diff
@@ -33,9 +38,9 @@ export interface Emitter<EPMap extends EventPayloadMap> {
  * @returns The event handler.
  */
 export const defineHandler = <
-  EPMap extends EventPayloadMap,
```

Comment:
Having the `EPMap extends EventPayloadMap` constraint makes it difficult to type `Emitter` and `EmitterEnv` correctly due to the circular reference